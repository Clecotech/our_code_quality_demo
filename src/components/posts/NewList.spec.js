import React from "react";
import { shallow, mount } from "enzyme";
import { NewList } from "./NewList";

describe("NewList", () => {
  it("renders empty by default", () => {
    const wrapper = shallow(<NewList onPostCreated={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("post creation", () => {
    describe("#onPostValueChanged fired", () => {
      it("unit: updates respective input value state", () => {
        const event = { target: { value: "Test Value" } };
        const wrapper = shallow(<NewList onPostCreated={jest.fn()} />);
        wrapper.instance().onPostValueChanged("title")(event);
        expect(wrapper.state("title")).toBe("Test Value");
      });
    });

    describe("title not provided", () => {
      it("displays error message", () => {
        // UH OH, test with state directly?
        const wrapper = mount(<NewList onPostCreated={jest.fn()} />);
        // Needed a mount here because instance can not be accessed in shallow()
        wrapper.find("input").at(1).instance().value = "Example Body";
        wrapper.find("button").simulate("click");
        expect(wrapper.state("errorMessage")).toBe("Name not provided");
      });
    });

    describe("body not provided", () => {
      it("displays error message", () => {
        const wrapper = mount(<NewList onPostCreated={jest.fn()} />);
        wrapper.setState({ title: "Example Title" });
        wrapper.find("button").simulate("click");
        expect(wrapper.state("errorMessage")).toBe("Email not provided");
      });
    });

    describe("valid input provided", () => {
      it("resets form values to empty", () => {
        const mockedEvent = jest.fn();
        const wrapper = mount(<NewList onPostCreated={mockedEvent} />);
        wrapper.setState({ title: "Example Title", body: "Example Body" });
        wrapper.find("button").simulate("click");
        expect(wrapper.state("errorMessage")).toBe("");

        const { title, body } = wrapper.state();
        expect(title).toBe("");
        expect(body).toBe("");
      });
    });
  });
});
