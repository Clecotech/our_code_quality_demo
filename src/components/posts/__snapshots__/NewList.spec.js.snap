// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewList renders empty by default 1`] = `
<ForwardRef(Container)>
  <ForwardRef(Grid)
    container={true}
  >
    <ForwardRef(Grid)
      className="Container"
      item={true}
      xs={12}
    >
      <ForwardRef(Box)
        className="TextBox"
      >
        <ForwardRef(TextField)
          fullWidth={true}
          label="Please Enter Your Name"
          onChange={[Function]}
          type="text"
          value=""
          variant="outlined"
        />
        <ForwardRef(TextField)
          fullWidth={true}
          label="Please Enter Your Email"
          onChange={[Function]}
          sx={
            Object {
              "marginTop": "1rem",
            }
          }
          type="text"
          value=""
        />
      </ForwardRef(Box)>
      <div
        className="btn"
      >
        <ForwardRef(Button)
          onClick={[Function]}
          sx={
            Object {
              "backgroundColor": "#222",
            }
          }
          variant="contained"
        >
          Create Post
        </ForwardRef(Button)>
      </div>
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</ForwardRef(Container)>
`;
